lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -2012,7 +2147,8 @@ class DeclAndTypePrinter::Implementation
 public:
   void print(Type ty, Optional<OptionalTypeKind> optionalKind,
              Identifier name = Identifier(),
-             IsFunctionParam_t isFuncParam = IsNotFunctionParam) {
+             IsFunctionParam_t isFuncParam = IsNotFunctionParam,
+             size_t *nameIndex = nullptr) {
```

Comment:
This doesn't make sense as an API. Please use optional, or maybe a signed type. But optional is probably best.

Reply:
I was also thinking, maybe we could compute this in the caller and just pass it as `name`.

Reply:
Another thought: this requires a bit of refactoring but `name` doesn't need to be an `Identifier` which might help. We could refactor it to be a `std::string` and in the caller just use `std::to_string`.